### PR TITLE
support for multicall in geth

### PR DESCRIPTION
**Pull request by KillariDev**
enables geth multicall
image

### CODE REVIEW

File: app/ts/background/settings.ts [hunk 1]
```diff
@@ -82,7 +82,6 @@ export const defaultRpcs: RpcEntries = [
 		minimized: true,
 		weth: 0x105083929bf9bb22c26cb1777ec92661170d4285n,
 	},
-	/*
 	{
 		name: 'Eth (geth-multi)',
```
Comment (MicahZoltu) [inline app/ts/background/settings.ts:86]: Can we rename this?  Perhaps `Ethereum (experimental geth)` so it is clear to users they probably shouldn't use it right now?

File: app/ts/simulation/services/EthereumClientService.ts [hunk 2]
```diff
@@ -189,12 +189,21 @@ export class EthereumClientService {
...
+				blockStateCalls: blockStateCalls,
+				traceTransfers: true,
+				validation: false,
+			},
+			blockTag === parentBlock.number + 1n ? blockTag - 1n : blockTag
+		] } as const
+		console.log(blockStateCalls)
 		console.log(JSON.stringify(ExecutionSpec383MultiCallParams.serialize(call)))
 		const unvalidatedResult = await this.requestHandler.jsonRpcRequest(call)
+		console.log(unvalidatedResult)
```
Comment (MicahZoltu) [inline app/ts/simulation/services/EthereumClientService.ts:206]: Remove debugging code.

File: app/ts/simulation/services/EthereumClientService.ts [hunk 3]
```diff
@@ -250,9 +259,9 @@ export class EthereumClientService {
 		}
 		const erc20ABI = ['event Transfer(address indexed from, address indexed to, uint256 value)']
 		const erc20 = new ethers.Interface(erc20ABI)
-		const flattenedLogs = events.flat() 
+		const flattenedLogs = events.flat()
 		const parsedEthLogs = parseEthLogs(flattenedLogs)
-		const addressesWithEthTransfers = new Set<bigint>(parsedEthLogs.map(log => log.args[0]).concat(parsedEthLogs.map(log => log.args[1]).concat(senders)))
+		const addressesWithEthTransfers = new Set<bigint>(parsedEthLogs.map(log => EthereumAddress.parse(log.args[0])).concat(parsedEthLogs.map(log => EthereumAddress.parse(log.args[1])).concat(senders)))
```
Comment (MicahZoltu) [inline app/ts/simulation/services/EthereumClientService.ts:264]: Consider extracting these lambdas out into named functions on a previous line for improved clarity:
```suggestion
		const extractLogTopic = (log: TodoTypeHere) => EthereumAddress.parse(log.args[0])
		const extractLogSender = (log: TodoTypeHere) => EthereumAddress.parse(log.args[1])
		const addressesWithEthTransfers = new Set<bigint>(parsedEthLogs.map(extractLogTopic).concat(parsedEthLogs.map(extractLogSender).concat(senders)))
```
I'm not sure I got these names right, they are meant as an example.

File: app/ts/simulation/services/EthereumClientService.ts [hunk 4]
```diff
@@ -321,7 +331,7 @@ export class EthereumClientService {
 						loggersAddress: log.address,
 						data: 'data' in log && log.data !== undefined ? log.data : new Uint8Array(),
 						topics: 'topics' in log && log.topics !== undefined ? log.topics : [],
-					})),
+					})).filter((x) => x.loggersAddress !== 0x0n), //TODO, keep eth logs
```
Comment (MicahZoltu) [inline app/ts/simulation/services/EthereumClientService.ts:334]: Create issue for TODO or address it in this PR.

**Review by MicahZoltu — CHANGES_REQUESTED** [review]
Minor things, but I would like to see them addressed or discussed before merge.

File: app/ts/simulation/services/EthereumClientService.ts (hunk 4 above)
Comment (KillariDev) [reply app/ts/simulation/services/EthereumClientService.ts:325]: created issue about it

File: app/ts/background/settings.ts [hunk 5]
```diff
@@ -82,17 +82,16 @@ export const defaultRpcs: RpcEntries = [
 		minimized: true,
 		weth: 0x105083929bf9bb22c26cb1777ec92661170d4285n,
 	},
-	/*
```
Comment (jubalm) [inline app/ts/background/settings.ts:85]: this code is just commented out, you might want to reference on another issue if this one needs to be redone later

**Review by jubalm — APPROVED** [review]

**Review by MicahZoltu — APPROVED** [review]

**Pull request merged**